python/lsst/analysis/tools/actions/vector/vectorActions.py

```diff
@@ -82,6 +115,36 @@ def __call__(self, data: KeyedData, **kwargs) -> Vector:
         return (vecA - vecB) / vecB
 
 
+class Sn(VectorAction):
+    """Compute signal-to-noise in the given flux type"""
+
+    fluxType = Field[str](doc="Flux type to calculate the S/N in.", default="{band}_psfFlux")
+    uncertaintySuffix = Field[str](
+        doc="Suffix to add to fluxType to specify uncertainty column", default="Err"
+    )
+    band = Field[str](doc="Band to calculate the S/N in.", default="i")
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield (fluxCol := self.fluxType), Vector
+        yield f"{fluxCol}{self.uncertaintySuffix}", Vector
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        """Computes S/N in self.fluxType
+        Parameters
+        ----------
+        df : `Tabular`
+        Returns
+        -------
+        result : `Vector`
+            Computed signal-to-noise ratio.
+        """
+        fluxCol = self.fluxType.format(**(kwargs | dict(band=self.band)))
+        errCol = f"{fluxCol}{self.uncertaintySuffix.format(**kwargs)}"
+        result = cast(Vector, data[fluxCol]) / data[errCol]  # type: ignore
+
+        return np.array(cast(Vector, result))
```

Comment:
Was it you I discussed this with before and it was resolved? (Sorry I have been looking at so many reviews recently)